apps/pie-docs/test/helpers/functions.js

```diff
@@ -0,0 +1,8 @@
+const isElementVisibleInViewport = (element) => element.evaluate((el) => {
+    const rect = el.getBoundingClientRect();
+    return rect.top >= 0 && rect.bottom <= window.innerHeight;
+});
+
+export {
+    isElementVisibleInViewport,
+};
```

Comment:
pretty sure Playwright provides this kind of utility. Have you seen anything similar on their docs?

Reply:
such as these: https://playwright.dev/docs/api/class-elementhandle#element-handle-is-visible

Reply:
I did yeah, so that checks if it's in the DOM, not if it's in the viewport. So the test will fail because it thinks that the header is still on the page.